src/components/TodoComponents/TodoForm.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import './Todo.css'
+
+class TodoForm extends React.Component{
+    constructor(props){
+        super(props);
+        this.taskInputRef = React.createRef();
+        this.state = {
+            task: ''
+        }
+    }
+
+    handleInput = (event)=>{
+        this.setState({
+            task: event.target.value
+        })
+    }
+
+    handleAddTodo = (event)=>{
+        event.preventDefault();
+        this.props.addTask(this.state.task);
+        this.setState({
+            task: ''
+        })
+        this.taskInputRef.current.value = '';
+    }
+
+    handleClearCompleted = (event)=>{
+        event.preventDefault();
+        this.props.clearCompleted();
+    }
+    
+    render(){
+        return(
+            <form className="todo-form">
```

Comment:
You can simplify this a little bit with an `onSubmit` property on your form element, calling a submit handler that takes care of both clicking the submit button or pressing enter. Then you'd only need an `onClick` property on the clear button to differentiate the two.